crategen/validators.py

```diff
@@ -0,0 +1,109 @@
+from pydantic import ValidationError
+from typing import Union
+from .models import WRROCProcess, WRROCWorkflow, WRROCProvenance
+from urllib.parse import urlparse
+
+def validate_wrroc(data: dict) -> Union[WRROCProvenance, WRROCWorkflow, WRROCProcess]:
+    """
+    Validate that the input data is a valid WRROC entity and determine which profile it adheres to.
+    
+    This function attempts to validate the input data against the WRROCProvenance model first.
+    If that validation fails, it attempts validation against the WRROCWorkflow model.
+    If that also fails, it finally attempts validation against the WRROCProcess model.
+    
+    Args:
+        data (dict): The input data to validate.
+    
+    Returns:
+        Union[WRROCProvenance, WRROCWorkflow, WRROCProcess]: The validated WRROC data, indicating the highest profile the data adheres to.
```

Comment:
Please remove type information from the docstring. We already have type hints. Using both can be a source of errors/confusion.

Reply:
I’ve removed type information from the docstrings as they are redundant due to the type hints already provided.